WcaOnRails/spec/helpers/notifications_helper_spec.rb

```diff
@@ -45,15 +48,27 @@
         notifications = helper.notifications_for_user(delegate)
         expect(notifications).to eq [
           {
-            text: "#{unconfirmed_competition.name} is not confirmed",
-            url: edit_competition_path(unconfirmed_competition),
-          },
-          {
             text: "#{user.email} has claimed WCA ID #{person.wca_id}",
             url: edit_user_path(user.id, anchor: "wca_id"),
           },
         ]
       end
+
+      context "have delegated competitions that are missing reports" do
+        let!(:past_competition_missing_report) { FactoryGirl.create :competition, :past, :visible, :confirmed, delegates: [delegate] }
+        let!(:past_competition_having_report) { FactoryGirl.create :competition, :past, :visible, :confirmed, :with_delegate_report, delegates: [delegate] }
+        let!(:future_competition) { FactoryGirl.create :competition, :future, :visible, :confirmed, delegates: [delegate] }
+
+        it "asks me to submit the reports" do
+          notifications = helper.notifications_for_user(delegate)
+          expect(notifications).to eq [
+            {
+              text: "The delegate report for #{past_competition_missing_report.name} has not been submitted.",
+              url: delegate_report_path(past_competition_missing_report),
+            },
+          ]
+        end
+      end
```

Comment:
This is the new spec, the rest is just some clean up moving `let!` calls to a separate context to prevent them from affecting all specs.